### ISSUE TITLE
Severe error during shutdown while processing unconfirmed txes

### ISSUE THREAD

**Issue by LeeSmet** [nbh-digital/goldchain#47]
```
Caught stop signal, quitting...
Closing explorer...
Closing block creator...
Closing wallet...
goroutine 133 [running]:
runtime/debug.Stack(0x91ae7c, 0xc0003e4cc0, 0xc0001c2048)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0xab
runtime/debug.PrintStack()
	/usr/lib/go/src/runtime/debug/stack.go:16 +0x34
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/build.Severe(0xc0001c2540, 0x2, 0x2)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/build/critical.go:27 +0x8c
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/transactionpool.(*TransactionPool).ProcessConsensusChange(0xc0003b6000, 0x55f7d6f9e26b3d5f, 0x9f1b971d30ceb5f5, 0x1b2bb6f9b4e8e6fc, 0x91d320d46b06fe00, 0x0, 0x0, 0x0, 0xc0003a2b60, 0x1, ...)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/transactionpool/update.go:166 +0xe24
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).readlockUpdateSubscribers(0xc000362000, 0x0, 0x0, 0x0, 0xc000d5b5c0, 0x1, 0x1)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/subscribe.go:87 +0x229
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).managedAcceptBlock(0xc000362000, 0x34404818a6872676, 0xd3408f5e0b573267, 0x3ac5342e3adb7965, 0x9be7409ba29a68a9, 0x5d455fc9, 0x52e, 0x0, 0x0, 0x0, ...)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/accept.go:304 +0x2ce
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).managedReceiveBlocks(0xc000362000, 0xe40960, 0xc000b40be0, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/synchronize.go:212 +0x6be
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway.(*Gateway).managedRPC(0xc000276380, 0xc0002b2520, 0x14, 0xcf61c0, 0xa, 0xc001950020, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway/rpc.go:66 +0x353
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway.(*Gateway).RPC(0xc000276380, 0xc0002b2520, 0x14, 0xcf61c0, 0xa, 0xc001950020, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway/rpc.go:76 +0xd2
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).threadedInitialBlockchainDownload.func2(0xc000362000, 0xc0001c3ec0, 0xc0001c3e00, 0xc00040c2e0, 0xc0001c3e18, 0xc0001c3e88, 0xc0001c3e08, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/synchronize.go:531 +0x1b0
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).threadedInitialBlockchainDownload(0xc000362000, 0xc00032efb8, 0x1)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/synchronize.go:561 +0x5e8
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).Start.func1(0xc000362000)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/consensusset.go:199 +0x67a
created by github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).Start
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/consensusset.go:191 +0x4d
Severe error: update consensus change in tx pool failed: error while updating txpool subscribers ThreadGroup already stopped
panic: Severe error: update consensus change in tx pool failed: error while updating txpool subscribers ThreadGroup already stopped


goroutine 133 [running]:
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/build.Severe(0xc0001c2540, 0x2, 0x2)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/build/critical.go:30 +0x128
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/transactionpool.(*TransactionPool).ProcessConsensusChange(0xc0003b6000, 0x55f7d6f9e26b3d5f, 0x9f1b971d30ceb5f5, 0x1b2bb6f9b4e8e6fc, 0x91d320d46b06fe00, 0x0, 0x0, 0x0, 0xc0003a2b60, 0x1, ...)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/transactionpool/update.go:166 +0xe24
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).readlockUpdateSubscribers(0xc000362000, 0x0, 0x0, 0x0, 0xc000d5b5c0, 0x1, 0x1)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/subscribe.go:87 +0x229
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).managedAcceptBlock(0xc000362000, 0x34404818a6872676, 0xd3408f5e0b573267, 0x3ac5342e3adb7965, 0x9be7409ba29a68a9, 0x5d455fc9, 0x52e, 0x0, 0x0, 0x0, ...)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/accept.go:304 +0x2ce
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).managedReceiveBlocks(0xc000362000, 0xe40960, 0xc000b40be0, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/synchronize.go:212 +0x6be
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway.(*Gateway).managedRPC(0xc000276380, 0xc0002b2520, 0x14, 0xcf61c0, 0xa, 0xc001950020, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway/rpc.go:66 +0x353
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway.(*Gateway).RPC(0xc000276380, 0xc0002b2520, 0x14, 0xcf61c0, 0xa, 0xc001950020, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/gateway/rpc.go:76 +0xd2
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).threadedInitialBlockchainDownload.func2(0xc000362000, 0xc0001c3ec0, 0xc0001c3e00, 0xc00040c2e0, 0xc0001c3e18, 0xc0001c3e88, 0xc0001c3e08, 0x0, 0x0)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/synchronize.go:531 +0x1b0
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).threadedInitialBlockchainDownload(0xc000362000, 0xc00032efb8, 0x1)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/synchronize.go:561 +0x5e8
github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).Start.func1(0xc000362000)
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/consensusset.go:199 +0x67a
created by github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus.(*ConsensusSet).Start
	/home/lee/go/src/github.com/nbh-digital/goldchain/vendor/github.com/threefoldtech/rivine/modules/consensus/consensusset.go:191 +0x4d
```

**Comment by LeeSmet** [comment]
Problem is that the wallet subscribes on the tpool, and if the wallet is being closed, but has not yet unsubscribed itself, it will simply return the error it gets when grabbing the thread from the threadpool. This error is not checked in the tpool.

**Comment by LeeSmet** [comment]
Fixed in b5cc902e1e30b58df78150f64fbb3dfc361db6d8
